### DISCUSSION TITLE
Question: Is it possible to propagate add_defines to dependencies?

### DISCUSSION THREAD

**Discussion by davidelahoz** [xmake-io/xmake#5756] [Q&A]
I'm trying to use sqlite in a xmake project. However, I need to increase the maximum number of columns allowed, which can be achieved by setting SQLITE_MAX_COLUMN macro during sqlite compilation. Unlickily, [sqlite3 recipe](https://github.com/xmake-io/xmake-repo/blob/master/packages/s/sqlite3/xmake.lua) doesn't provide an option for setting that macro value

Is there any way to propogate an add_dependency set in your project xmake.lua, so that it will be used when building a dependency?

(I know that databases shouldn't have that many columns, but I'm just using sqlite as a way to reduce the amount of storage consumed by a dataset with many features. No operations will be performed in the database, other that loading the data and then reading it into a pandas dataframe)

**Comment by waruqi** [comment]
you can rewite sqlite package to set it.
